src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedDayNames.cs

```diff
@@ -2,36 +2,93 @@
+
+            // Does not clone in setter, only in getter.
+            value[0] = null;
+            Assert.NotSame(value, format.AbbreviatedDayNames);
+            Assert.Equal(value, format.AbbreviatedDayNames);
         }
 
         [Fact]
-        public void AbbreviatedDayNames_Set_Invalid()
+        public void AbbreviatedDayNames_SetNulValue_ThrowsArgumentNullException()
         {
-            AssertExtensions.Throws<ArgumentNullException>("value", () => new DateTimeFormatInfo().AbbreviatedDayNames = null); // Value is null
-            AssertExtensions.Throws<ArgumentNullException>("value", () => new DateTimeFormatInfo().AbbreviatedDayNames = new string[] { "1", "2", "3", null, "5", "6", "7" }); // Value has null
-            AssertExtensions.Throws<ArgumentException>("value", (() => new DateTimeFormatInfo().AbbreviatedDayNames = new string[] { "sun" })); // Value.Length is not 7
+            var format = new DateTimeFormatInfo();
+            AssertExtensions.Throws<ArgumentNullException>("value", () => format.AbbreviatedDayNames = null);
+        }
+
+        [Fact]
+        public void AbbreviatedDayNames_SetNulValueInValue_ThrowsArgumentNullException()
+        {
+            var format = new DateTimeFormatInfo();
+            AssertExtensions.Throws<ArgumentNullException>("value", () => format.AbbreviatedDayNames = new string[] { "1", "2", "3", null, "5", "6", "7" });
+        }
 
-            // DateTimeFormatInfo.InvariantInfo is read only
+        public static IEnumerable<object[]> AbbreviatedDayNames_SetInvalidLength_TestData()
+        {
+            yield return new object[] { new string[] { "Sun" } };
+            yield return new object[] { new string[] { "Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Additional" } };
+        }
+
+        [Theory]
+        [MemberData(nameof(AbbreviatedDayNames_SetInvalidLength_TestData))]
+        public void AbbreviatedDayNames_SetInvalidLength_ThrowsArgumentException(string[] value)
+        {
+            var format = new DateTimeFormatInfo();
+            AssertExtensions.Throws<ArgumentException>("value", (() => format.AbbreviatedDayNames = value));
+        }
+
+        [Fact]
+        public void AbbreviatedDayNames_SetReadOnly_ThrowsInvalidOperationException()
+        {
             Assert.Throws<InvalidOperationException>(() => DateTimeFormatInfo.InvariantInfo.AbbreviatedDayNames = new string[] { "1", "2", "3", "4", "5", "6", "7" });
         }
+
+        [Fact]
+        public void AbbreviatedDayNames_FormatWithNull_ThrowsNullReferenceException()
+        {
+            var value = new string[] { "1", "2", "3", "4", "5", "6", "7" };
+            var format = new DateTimeFormatInfo
+            {
+                AbbreviatedDayNames = value
+            };
+            value[0] = null;
+
+            var dateTime = new DateTime(2014, 5, 28);
+            Assert.Throws<NullReferenceException>(() => dateTime.ToString("ddd MMM yy", format));
+        }
```

Comment:
I suggest we don't need a test validating that we null ref here and in similar places.  The setter does its best to validate that the input is valid, but cloning would be unnecessarily expensive, and if you explicitly then break it by setting an element of the array to null, that's on you.